src/background.ts

```diff
@@ -91,6 +93,22 @@ class ShortExt {
     launch() {
         this.interceptRequests();
     }
+
+    // Function to handle extension clicks
+    // Required: browser_action clause in manifest.json to tell the browser that we will need it's events.
+    redirectToBaseURI = (base_uri: String) => {
+        // Event Listener
+        chrome.browserAction.onClicked.addListener(function(tab) 
+            { 
+                // Check if the current tab has some url/event should not be triggered on empty tabs
+                if(tab.hasOwnProperty("url")){
```

Comment:
```suggestion
                if(isEmptyTab(tab)) {
                  return;
                }
```

Reply:
```typescript
isEmptyTab(tab) {
  return !tab.hasOwnProperty("url");
}
```

Please add `isEmptyTab` method after the refactor.
In general, we prefer readable variable & function names over comments. Comments are added when there are hidden knowledge that looks confusing to the readers.